typescript/cli/src/rebalancer/strategy/MinAmountStrategy.ts

```diff
@@ -0,0 +1,76 @@
+import type { ChainMap } from '@hyperlane-xyz/sdk';
+
+import type { RawBalances } from '../interfaces/IStrategy.js';
+
+import { BaseStrategy, type Delta } from './BaseStrategy.js';
+
+/**
+ * Configuration for minimum amount strategy
+ */
+export type MinAmountStrategyConfig = {
+  minAmount: bigint;
+  buffer: bigint;
+};
+
+/**
+ * Strategy implementation that rebalance based on minimum amounts
+ * It ensures each chain has at least the specified minimum amount
+ */
+export class MinAmountStrategy extends BaseStrategy {
+  private readonly config: ChainMap<MinAmountStrategyConfig>;
+
+  constructor(config: ChainMap<MinAmountStrategyConfig>) {
+    const chains = Object.keys(config);
+    super(chains);
+
+    for (const chain of chains) {
+      const { buffer, minAmount } = config[chain];
+
+      if (minAmount < 0n) {
+        throw new Error('Minimum amount cannot be negative');
+      }
+
+      if (buffer < 0n || buffer > 10_000n) {
+        throw new Error('Buffer must be between 0 and 10,000 basis points');
+      }
+    }
+
+    this.config = config;
+  }
+
+  /**
+   * Gets balances categorized by surplus and deficit based on minimum amounts and buffer
+   * Buffer (in basis points) is a bigint between 1n (0.01%) and 10_000n (100.00%), default 0n
+   */
+  protected getCategorizedBalances(rawBalances: RawBalances): {
+    surpluses: Delta[];
+    deficits: Delta[];
+  } {
+    return this.chains.reduce(
+      (acc, chain) => {
+        const minAmount = this.config[chain].minAmount;
+        const effectiveMin =
+          (minAmount * (10_000n + this.config[chain].buffer)) / 10_000n;
+        const balance = rawBalances[chain];
+
+        // If balance is less than effectiveMin, it has a deficit
+        if (balance < effectiveMin) {
```

Comment:
Should this be `balance < minAmount`?

Reply:
The effectiveMin already has the buffer, minAmount indicates the actual minimum

Reply:
Depends, I treated the min with the buffer applied as the "new min", so if you're below it, you should consider increasing.

What you're proposing is "if I go below a min amount, I should go up to min amount + buffer"?

Reply:
Right! Because if the effectiveMin is the threshold, doesn't it defeat the purpose of the minValue?

Reply:
Going "up to min amount + buffer" is what prevents the collateral being unbalanced just from losing 1 wei

Reply:
I think it makes more sense now that I re-read this

Reply:
Imagine the min is 100, I'm at 90, go back to 100 from rebalancing, any trade, no matter how small will just take me back to red